### PR TITLE
Add `runMaestro.ts` script 

### PR DESCRIPTION
**Pull request by itayplav**
### CHANGELOG

 none

### Dependencies

 none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [ ] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204680683447147

### CODE REVIEW

**Review by paullinator — CHANGES_REQUESTED** [review]
This could be simplified and less error prone by

1. Using clipanion
2. Break up the test specification like so

```
yarn maestro --tests [groupId]:[tagId],[groupIdOnly] --testNum 300-400,500-600 --skipTests [groupId] --skipNum 300,500-513
```

File: scripts/runMaestro.ts [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+  const { includedTests } = options
+  if (includedTests === '') {
+    console.log(
+      `
+    ${yellow} No test IDs or tags were provided.
+    ${bright}${green}Running all tests in 3 seconds.${reset}
+`
+    )
+    setTimeout(async () => {
+      await runMaestro({ ...options, includedTests: '0:200913' })
```
[the remark this answers is not in the document]
Comment (itayplav) [reply scripts/runMaestro.ts]: Yep this was arbitrary - v2 uses the [catch all](https://github.com/EdgeApp/edge-react-gui/pull/4206/commits/b2fbe701fa13a91e6f6047a72144322645ca317a#diff-857a35d1cacd08cfc4c49807a5589e430cf880c0ecdef0b65dab379e5aa176efR176) test selector when no arguments are specified i.e. `yarn maestro`

File: scripts/runMaestro.ts [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+type Tag = Record<string, Set<string>>
+type TagRegex = Record<string, Set<RegExp>>
+
+const green = '\x1b[32m'
+const reset = '\x1b[0m'
+const bright = '\x1b[1m'
+const yellow = '\x1b[33m'
+
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+class Maestro {
```
[the remark this answers is not in the document]
Comment (itayplav) [reply scripts/runMaestro.ts]: Class structure has been replaced w/ loose functions

File: scripts/runMaestro.ts [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+  --skip <Tests>          Skip tests with these IDs or tags. See examples below for more details.
+  --include-skipped       Skip tests that were both included and excluded. Overrides the default behavior of including tests.
+
+Examples:
+
+  # Run test ID 314 and 315. Use a comma to separate multiple IDs / ID ranges / tags.
+    ${green}yarn maestro 314,315${reset}
+
+  # Run all tests in the "login" test group
+    ${green}yarn maestro :login${reset}
```
[the remark this answers is not in the document]
Comment (itayplav) [reply scripts/runMaestro.ts]: This was an error in the documentation - now an empty string before/after a `:` is invalid, and at minimum requires a wildcard i.e. 

`+:2fa` -> all groups with the 2fa tag
`login:+` -> all tests in the login group

File: scripts/runMaestro.ts [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+  } else {
+    await runMaestro(options)
+  }
+}
+const args = process.argv.slice(2)
+if (args.includes('--help')) {
+  printHelp()
+} else {
+  const options: MaestroOptions = {
+    includedTests: args[0] ?? '',
```
[the remark this answers is not in the document]
Comment (itayplav) [reply scripts/runMaestro.ts]: Now uses [cac](https://github.com/cacjs/cac) for argument parsing

**Review by paullinator — APPROVED** [review]

**Pull request merged**